### ISSUE TITLE
The Products Widget Has A Site Breaking Glitch

### ISSUE THREAD

**Issue by Jacob-Bayless** [elementor/elementor#16001]
## Prerequisites

- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x ] The issue still exists against the latest stable version of Elementor.

## Description
The Products widget for Woocommerce displays strange sizes that shrink randomly when changing the column and row numbers. It is very erratic and not consistent. Sometimes even just clicking onto different tabs like Style or Advanced can cause this issue to appear.

## Steps to reproduce

1. Add a Products widget.
<img width="1658" alt="Add widget" src="https://user-images.githubusercontent.com/64002790/129658920-c4994eb8-02c8-479b-b976-cbe7182c1e58.png">

2. Change columns to 5 and rows to 1.
<img width="1672" alt="Screen Shot 2021-08-16 at 10 22 31 PM" src="https://user-images.githubusercontent.com/64002790/129659162-afc68210-24e0-4eeb-8561-aa4007242707.png">

3. Changing the columns and rows back to the default numbers will sometimes revert the size back to normal. But other times it will remain broken (See below)
<img width="1680" alt="Screen Shot 2021-08-16 at 10 23 29 PM" src="https://user-images.githubusercontent.com/64002790/129659321-fdde2c8e-2b4b-4ac0-9c76-e0c1f6583448.png">

4. Changing the columns and rows to other numbers can also sometimes fix the issue but not always.

Example 1
<img width="1680" alt="Screen Shot 2021-08-16 at 10 23 37 PM" src="https://user-images.githubusercontent.com/64002790/129659902-071f0675-95ea-436e-ba9d-480af5ce9883.png">

Example 2
<img width="1680" alt="Screen Shot 2021-08-16 at 10 23 44 PM" src="https://user-images.githubusercontent.com/64002790/129659942-3864b718-3e25-4a68-8c66-ed9e6fa66824.png">

This issue also causes mobile sites to have sizing issues based on the amount of columns.
1. Add Product Widget with 10 rows
<img width="1680" alt="Screen Shot 2021-08-16 at 10 49 51 PM" src="https://user-images.githubusercontent.com/64002790/129660853-503d0272-9112-4e57-b23c-732769b822c6.png">

2. Change the rows to 200
<img width="1680" alt="Screen Shot 2021-08-16 at 10 50 49 PM" src="https://user-images.githubusercontent.com/64002790/129660887-b3f3be97-e449-4ece-bf47-89da676b8efb.png">

<img width="1680" alt="Screen Shot 2021-08-16 at 10 50 58 PM" src="https://user-images.githubusercontent.com/64002790/129660906-b2e48398-3e8d-4933-8f5f-45a3e95024af.png">

I know this could be worked around using pagination, but this particular client wants all the products in each category to be on one continuous scrolling page.

Ecommerce sites using the Products plugin for their main product page templates will be devastated until this issue is fixed. I say that because I've also seen just refreshing the page cause this issue to take effect.

## Isolating the problem

- [x ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x ] This bug happens with a default WordPress theme active.
- [x ] I can reproduce this bug consistently using the steps above.

## Environment
				
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Percona Server (GPL), Release rel5.7, Revision 26 v5.7.26-29
	PHP Version: 7.3.12
	PHP Max Input Vars: 1000
	PHP Max Post Size: 100M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.8
	Site URL: https://a93.e01.myftpupload.com
	Home URL: https://a93.e01.myftpupload.com
	WP Multisite: No
	Max Upload Size: 100 MB
	Memory limit: 256M
	Permalink Structure: /%year%/%monthnum%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.4.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15

== Active Plugins ==
	Elementor
		Version: 3.4.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.3.7
		Author: Elementor.com

	Enhanced Media Library
		Version: 2.8.7
		Author: wpUXsolutions

	JetWoo Widgets For Elementor
		Version: 1.1.2
		Author: Crocoblock

	Premium Addons for Elementor
		Version: 4.4.7
		Author: Leap13

	Search Engine Visibility
		Version: 0.5
		Author: GoDaddy

	ShareThis Dashboard for Google Analytics
		Version: 2.5.2
		Author: ShareThis

	UpdraftPlus - Backup/Restore
		Version: 1.16.59
		Author: UpdraftPlus.Com, DavidAnderson

	WooCommerce
		Version: 5.5.2
		Author: Automattic

	Yoast SEO
		Version: 16.9
		Author: Team Yoast

== Must-Use Plugins ==
	System Plugin
		Version: 4.10.1
		Author: 

== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified
	JetWoo Widgets For Elementor: Compatibility not specified
	Premium Addons for Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	

== Elements Usage ==
	
	footer : 1
		heading : 5
		icon-list : 4
		image : 1
		social-icons : 1
		text-editor : 1
	header : 1
		button : 1
		image : 2
		nav-menu : 1
		woocommerce-menu-cart : 2
	popup : 2
		icon : 1
		nav-menu : 1
	wp-page : 1
		button : 13
		heading : 18
		image : 4
		image-box : 2
		jet-woo-widgets-products : 8
		slides : 1
		star-rating : 2
		text-editor : 6
	product-archive : 1
		jet-woo-widgets-products : 1
		slides : 1
		woocommerce-products : 2

== Log ==
	
PHP: showing 3 of 3PHP: 2021-07-08 01:56:08 [notice X 6][/var/www/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-07-23 03:08:19 [error X 1][/var/www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::285] Uncaught Error: Call to a member function update_meta() on bool in /var/www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:285
Stack trace:
#0 /var/www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(166): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(NULL, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /var/www/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /var/www/wp-includes/class-wp-hook.php(303): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /var/www/wp-includes/class-wp-hook.php(327): WP_Hook->apply_filters('', Array)
#5 /var/www/wp-includes/plugin.php(470): WP_Hook->do_action(Array)
#6 /var/www/wp-admin/admin-ajax.php(187): d [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-08-13 02:08:54 [notice X 1][/var/www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::166] Undefined index: editor_post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 13 of 132021-08-04 03:46:52 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.5',
  'to' => '3.3.1',
)]
2021-08-04 03:46:52 [info] elementor::elementor_updater Started 
2021-08-04 03:46:52 [info] Elementor/Upgrades - _on_each_version Start  
2021-08-04 03:46:54 [info] Elementor/Upgrades - _on_each_version Finished 
2021-08-04 03:46:54 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.2.5',
  'to' => '3.3.1',
)]
2021-08-04 03:51:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.3.1',
  'to' => '3.3.5',
)]
2021-08-13 01:20:42 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.3.5',
  'to' => '3.3.6',
)]
2021-08-17 00:13:25 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.3.6',
  'to' => '3.3.7',
)]
2021-08-17 00:13:44 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.3.1',
  'to' => '3.4.0',
)]
2021-08-17 00:13:45 [info] elementor::elementor_updater Started 
2021-08-17 00:13:45 [info] Elementor/Upgrades - _on_each_version Start  
2021-08-17 00:13:46 [info] Elementor/Upgrades - _on_each_version Finished 
2021-08-17 00:13:46 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.3.1',
  'to' => '3.4.0',
)]

**Comment by ATaylor66** [comment]
I've been trying to figure out the same issue and found that woocommerce-smallscreen.css was set like this:

 `.woocommerce ul.products[class*=columns-] li.product,
.woocommerce-page ul.products[class*=columns-] li.product {
 **width:48%;**
 float:left;
 clear:both;
 margin:0 0 2.992em
}`

So I added custom css to the customiser and it seems to have fixed it for now. I don't know if there was a better way around it though.

`.woocommerce ul.products[class*=columns-] li.product,
.woocommerce-page ul.products[class*=columns-] li.product {
 width:100%;
}`

**Comment by d19dotca** [comment]
I'm having the same issue today I noticed when I modified the row value.

**Comment by nataib** [comment]
I'm having the same issue with the Products widget since updating to Elementor Pro 3.3.6 and 3.3.7. The updates have made the products tiny and have pushed it to the left side of the screen. I've tried decreasing the product gaps, but it distorts it even more. The problem I have is site wide, doesn't matter how many rows or columns in the rows.

Here's what it looks like when I've updated to 3.3.6 and 3.3.7:
<img width="1186" alt="Screen Shot 2021-08-11 at 8 59 53 AM" src="https://user-images.githubusercontent.com/89171666/129992347-94d6077a-5474-4006-a0e9-370897a3f5d3.png">


Here's what it should look like (I reverted back to Elementor Pro 3.3.2.):
<img width="1239" alt="Screen Shot 2021-08-11 at 9 23 41 AM" src="https://user-images.githubusercontent.com/89171666/129992459-6df9bbed-c8bd-461c-a636-3c36e9647fe4.png">

**Comment by enuzzo** [comment]
Same exact issue here. Live site: https://erbeticino.ch

**Desktop:** working as usual.
**Tablet, Mobile:** grid problem > extra margins/padding for each product.
 
![IMG_B8CD7D7C7061-1](https://user-images.githubusercontent.com/3628008/130037184-7f967694-2d9e-4a8c-beda-03fa8bd9fec1.jpeg)

**Comment by DCMadmin** [comment]
Fault is in Elmentor Pro 3.3.6 & 3.3.7. Works fine in 3.3.5.

**Comment by DCMadmin** [comment]
@nickgreen posted this useful info in the ticket #16067 that doesn’t appear in here I believe.

I have temporarily patched the front end by adding the following CSS:

.elementor-products-grid ul.products li.product { width: auto !important; padding: 0 !important; margin: 0 !important; float: none !important; }

The items are not formatted completely correctly but they are readable.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v3.3.8

Feel free to update

Thanks!
